File: dcp-solutions/2024-12-19/solution.php
```php
<?php
function minSteps($points) {
    $x = $points[0][0];
    $y = $points[0][1];
    $dx = 0;
    $dy = 0;

    foreach ($points as $point) {
        list($px, $py) = $point;
        if ($px != $x || $py != $y) {
            $dx = $px - $x;
            $dy = $py - $y;
            break;
        }
    }

    $steps = abs($dx) + abs($dy);
    foreach ($points as $point) {
        list($px, $py) = $point;
        if ($dx != 0 && $px < $x) {
            $steps += min(max(0, $x - $px), abs($dx));
            $x = $px;
        } elseif ($dy != 0 && $py > $y) {
            $steps += min(max(0, $py - $y), abs($dy));
            $y = $py;
        }
    }

    return $steps;
}
```

File: dcp-solutions/2024-12-19/solution.ex
```elixir
def calculate_steps(points) do
  points = Enum.map(points, fn point -> [point] end)
  |> Enum.map(&get_distance/1)
  |> List.reduce(0, &add/2)

  def get_distance([x1, y1 | tail]), do: abs(x1 - head(tail)) + abs(y1 - head(tail))
  def get_distance([x1, y1]), do: abs(x1) + abs(y1)
  def get_distance([]), do: 0

  def add(num1, num2), do: num1 + num2
end
```